### PR TITLE
work-around joinpath error when cloning git ssh urls (fix #20695)

### LINKED ISSUE #20695: Cloning from GitHub over ssh in 0.6-dev
I get the following error when trying to clone an unregistered package:

```
julia> Pkg.clone("git@github.com:krcools/CollisionDetection.git")
ERROR: ArgumentError: drive mismatch: METADATA git@github.com:krcools/CollisionDetection.git
Stacktrace:
 [1] joinpath(::String, ::String) at .\path.jl:197
 [2] clone(::String) at .\pkg\entry.jl:210
 [3] (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#clone,Tuple{String}})() at .\pkg\dir.jl:31
 [4] cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#clone,Tuple{String}}, ::String) at .\file.jl:58
 [5] #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at .\pkg\dir.jl:31
 [6] clone(::String) at .\pkg\pkg.jl:154

julia> versioninfo()
Julia Version 0.6.0-dev.2635
Commit dc2459d* (2017-02-13 07:46 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) CPU E3-1505M v5 @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)

julia> Pkg.clone("https://github.com/krcools/CollisionDetection.git")
INFO: Cloning CollisionDetection from https://github.com/krcools/CollisionDetection.git
INFO: Computing changes...
INFO: No packages to install, update or remove
```

As you can see, cloning works over https but not over ssh. Somehow the ssh url (which contains a colon and thus can be mistaken for a drive/path pair) ends up in `joinpath`, with an `ArgumentError` thrown as result.

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

File: base/pkg/entry.jl [hunk 1]
```diff
@@ -207,18 +207,20 @@ function clone(url::AbstractString, pkg::AbstractString)
 end
 
 function clone(url_or_pkg::AbstractString)
+    if !(':' in url_or_pkg)
     urlpath = joinpath("METADATA",url_or_pkg,"url")
```
Comment (tkelman) [inline base/pkg/entry.jl]: this and every other unmodified line in the function needs re-indenting

**Comment by omus** [comment]
Would be easy to test you separate the logic. Maybe make a `pkg_name(url_or_pkg::AbstractString)` which always returns the package name?

File: base/pkg/entry.jl [hunk 2]
```diff
@@ -206,21 +206,20 @@ function clone(url::AbstractString, pkg::AbstractString)
     end
 end
 
-function clone(url_or_pkg::AbstractString)
-    urlpath = joinpath("METADATA",url_or_pkg,"url")
-    if !(':' in url_or_pkg) && isfile(urlpath)
-        pkg = url_or_pkg
-        url = readchomp(urlpath)
-        # TODO: Cache.prefetch(pkg,url)
```
Comment (tkelman) [inline base/pkg/entry.jl]: this todo is still relevant

Comment (StefanKarpinski) [inline base/pkg/entry.jl]: Then it should be added back with more explanation because I had no idea what it meant and I originally wrote this code.

Comment (StefanKarpinski) [inline base/pkg/entry.jl]: Also, we're retiring this code soon and we've been living without this for how long now? I think it's fine.

**Pull request merged**